lib/provisioning/Provisioner.js

```diff
@@ -805,7 +802,7 @@ Provisioner.prototype.unlink = Promise.coroutine(function*(options) {
             throw new Error(s);
         }
         else {
-            log.error(err);
+            // err is useless, change the message and throw
```

Comment:
:cry: Poor error, crying out in the hopes that someone is listening, and yet... nothing.

What happens if `this._unlinkValidator.validate(options);` throws a `TypeError`? Will we ever know? 0:)